### PR TITLE
Silent gtk action 2

### PR DESCRIPTION
**Pull request by raveit65**
This make the build log finally readable.
Please review.

### CODE REVIEW

**Comment by sc0w** [comment]
see https://github.com/mate-desktop/caja/pull/1243

please use `G_GNUC_BEGIN_IGNORE_DEPRECATIONS` and `G_GNUC_END_IGNORE_DEPRECATIONS` with semicolon to avoid a lot of new cppcheck warnings

I can do the job here if you are lazy :)

**Comment by raveit65** [comment]
Go ahead if you have already check running caja with that.
I have no objection if it fix ccpcheck warnings and is not a bug by ccpcheck :)

**Comment by raveit65** [comment]
Hmm, you did it in an extra commit.
This is suboptimal for fixing previous commits, because i found an issue with the seachbar.
I think i will do that for myself.

**Comment by sc0w** [comment]
no problem

btw, I notice some code like this:

```
 else
    G_GNUC_END_IGNORE_DEPRECATIONS;
    {
```

it probably is wrong and it needs to put in other place

**Comment by raveit65** [comment]
yes, i found it already. Using this looks weird but it works.
```
diff -uprN caja-1.23.0.orig/src/caja-navigation-window-menus.c caja-1.23.0/src/caja-navigation-window-menus.c
--- caja-1.23.0.orig/src/caja-navigation-window-menus.c	2019-06-28 12:06:17.893622000 +0200
+++ caja-1.23.0/src/caja-navigation-window-menus.c	2019-06-28 12:10:46.904662493 +0200
@@ -715,7 +715,6 @@ action_show_hide_search_callback (GtkAct
         caja_navigation_window_show_search (window);
     }
     else
-    G_GNUC_END_IGNORE_DEPRECATIONS;
     {
         CajaWindowSlot *slot;
         GFile *location = NULL;
@@ -740,8 +739,10 @@ action_show_hide_search_callback (GtkAct
                 uri = caja_query_get_location (query);
                 if (uri != NULL)
                 {
+                    G_GNUC_END_IGNORE_DEPRECATIONS;
                     location = g_file_new_for_uri (uri);
                     g_free (uri);
+                    G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
                 }
                 g_object_unref (query);
             }
@@ -750,7 +751,9 @@ action_show_hide_search_callback (GtkAct
         /* Last try: use the home directory as the return location */
         if (location == NULL)
         {
+            G_GNUC_END_IGNORE_DEPRECATIONS;
             location = g_file_new_for_path (g_get_home_dir ());
+            G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
         }
 
         caja_window_go_to (CAJA_WINDOW (window), location);
@@ -758,6 +761,7 @@ action_show_hide_search_callback (GtkAct
 
         caja_navigation_window_hide_search (window);
     }
+    G_GNUC_END_IGNORE_DEPRECATIONS;
 }
 
 static void
```
I exclude other external functions.

**Comment by raveit65** [comment]
I added missing semicolon and rebased PR.

File: src/caja-navigation-window-menus.c [hunk 1]
```diff
@@ -689,11 +709,13 @@ action_show_hide_search_callback (GtkAction *action,
...
     window = CAJA_NAVIGATION_WINDOW (user_data);
 
+    G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
     if (gtk_toggle_action_get_active (GTK_TOGGLE_ACTION (action)))
     {
         caja_navigation_window_show_search (window);
     }
     else
+    G_GNUC_END_IGNORE_DEPRECATIONS;
     {
```
Comment (sc0w) [inline src/caja-navigation-window-menus.c]:
```
    else
    G_GNUC_END_IGNORE_DEPRECATIONS;
    {
```

it seems to be wrong

File: src/file-manager/fm-icon-view.c [hunk 2]
```diff
@@ -1596,8 +1602,10 @@ action_reversed_order_callback (GtkAction *action,
 
     icon_view = FM_ICON_VIEW (user_data);
 
+    G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
     if (set_sort_reversed (icon_view,
                            gtk_toggle_action_get_active (GTK_TOGGLE_ACTION (action))))
+    G_GNUC_END_IGNORE_DEPRECATIONS;
     {
```
Comment (sc0w) [inline src/file-manager/fm-icon-view.c]:
```
if (......)
G_GNUC_END_IGNORE_DEPRECATIONS;
{
```

it seems to be wrong

File: src/caja-navigation-window-menus.c (hunk 1 above)
Comment (raveit65) [reply src/caja-navigation-window-menus.c]: I know, i am thinking about to make at better as here (already posted).
https://github.com/mate-desktop/caja/pull/1242#issuecomment-506695298

File: src/file-manager/fm-icon-view.c (hunk 2 above)
Comment (raveit65) [reply src/file-manager/fm-icon-view.c]: better?

File: src/caja-navigation-window-menus.c (hunk 1 above)
Comment (raveit65) [reply src/caja-navigation-window-menus.c]: last commit fixes caja-search.
I hope ccpcheck tool is happy.

**Comment by raveit65** [comment]
Caja-search is working again with last commit. I will squash that commit when it is approved.

File: src/caja-navigation-window-menus.c [hunk 3]
```diff
@@ -735,7 +759,11 @@ action_show_hide_search_callback (GtkAction *action,
         g_object_unref (location);
 
         caja_navigation_window_hide_search (window);
+    /* needed to exclude everything inside the `else` */
+    G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
     }
+    G_GNUC_END_IGNORE_DEPRECATIONS;
+    /*end deprecation check*/
```
Comment (sc0w) [inline src/caja-navigation-window-menus.c]:
```
    G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
    }
    G_GNUC_END_IGNORE_DEPRECATIONS;
```

I don't understand this change, it seems to ignore nothing, I think it isn't needed.

If you like just to ignore the warning of the line `if (gtk_toggle_action_get_active (GTK_TOGGLE_ACTION (action)))`

I suggest just to use a gboolean variable, I think it is more clear and readable:

```
gboolean variable;
...
...
...
G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
variable = gtk_toggle_action_get_active (GTK_TOGGLE_ACTION (action));
G_GNUC_END_IGNORE_DEPRECATIONS;

if (variable)
```

Comment (raveit65) [reply src/caja-navigation-window-menus.c]: This bracket is the ending from the else condidition. Without this caja-searchbar don't work.

Comment (raveit65) [reply src/caja-navigation-window-menus.c]: Is your suggestion tested?

Comment (sc0w) [reply src/caja-navigation-window-menus.c]: I didn't tested but it must work, you can believe me :)

Comment (raveit65) [reply src/caja-navigation-window-menus.c]: Done, and it works , thx.

**Comment by raveit65** [comment]
Thanks for the review, i don't have a fetish to see warnings which can't be fixed without removing a typical MATE look and feel, and i prefer not overseen important warnings.
I have seen them since 2011/12, enough is enough :)
Go ahead with merging https://github.com/mate-desktop/caja/pull/1241

**Pull request merged**